data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -165,6 +170,7 @@ private void flushBatch(final BulkRequest bulkRequest) {
     bulkRequestTimer.record(() -> {
       try {
         bulkRetryStrategy.execute(bulkRequest);
+        bulkRequestSizeBytes.addAndGet(bulkRequest.estimatedSizeInBytes());
```

Comment:
Can you explain why an AtomicLong is required here? Why wouldn't it be thread safe to just increment a Counter?

Reply:
I saw all other places using `gauge` also uses AtomicLong, and it seems to be safer since we do have many threads. Now we are using Summary, we don't have this concern any longer.